### PR TITLE
feat(Docker): add Secrets, Compose v2, Supply Chain Security, Podman

### PR DESCRIPTION
**Pull request by otkd**
- Updates Docker Compose references to use v2 `docker compose` command
- Change ICC recommendation to use network policies  instead of disabling
- Following existing syntax adds Rule 13 covering Docker Secrets
- Merges the linting and container scanning section into a single rule
- Fixes issue in proposed solution in Rule 10 doesn’t actually help implement the suggestion
- Expand rootless mode
- Add high level practices covering Docker supply chain security
- Add Podman as an alternative to Docker for secure defaults

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [x] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [x] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [x] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [x] All your assets are stored in the **assets** folder.
- [x] All the images used are in the **PNG** format.
- [x] Any references to websites have been formatted as `[TEXT](URL)`
- [x] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [x] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/Docker_Security_Cheat_Sheet.md [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
 
 ## Rules
 
 ### RULE \#0 - Keep Host and Docker up to date
 
-To prevent from known, container escapes vulnerabilities, which typically end in escalating to root/administrator privileges, patching Docker Engine and Docker Machine is crucial.
+To prevent known container escapes vulnerabilities, which typically result in the attacker gaining root access to the host, it is crucial to keep the host and Docker up to date. This includes the host kernel and Docker Engine.
 
-In addition, containers (unlike in virtual machines) share the kernel with the host, therefore kernel exploits executed inside the container will directly hit host kernel. For example, kernel privilege escalation exploit ([like Dirty COW](https://github.com/scumjr/dirtycow-vdso)) executed inside a well-insulated container will result in root access in a host.
+This is due to the fact that containers share the host's kernel. If the host's kernel is vulnerable, the containers are also vulnerable. For example, the kernel privilege escalation exploit ([Dirty COW](https://github.com/scumjr/dirtycow-vdso)) executed inside a well-insulated container would still result in root access on a vulnerable host.
```
Comment (szh) [inline cheatsheets/Docker_Security_Cheat_Sheet.md:15]:
```suggestion
This is due to the fact that containers share the host's kernel. If the host's kernel is vulnerable, the containers are also vulnerable. For example, the kernel privilege escalation exploit [Dirty COW](https://github.com/scumjr/dirtycow-vdso) executed inside a well-insulated container would still result in root access on a vulnerable host.
```
Removed parentheses around Dirty Cow link

File: cheatsheets/Docker_Security_Cheat_Sheet.md [hunk 2]
```diff
@@ -109,11 +109,11 @@ spec:
 
 As a Kubernetes cluster administrator, you can configure it using [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
 
-### RULE \#4 - Add –no-new-privileges flag
+### RULE \#4 - Prevent in container privilege escalation
```
Comment (szh) [inline cheatsheets/Docker_Security_Cheat_Sheet.md:112]:
```suggestion
### RULE \#4 - Prevent in-container privilege escalation
```
I'm no grammar expert but I think it's easier to read with a hyphen. It makes it clear that "in container" is one entity.

File: cheatsheets/Docker_Security_Cheat_Sheet.md [hunk 3]
```diff
@@ -133,14 +133,13 @@ spec:
 
 As a Kubernetes cluster administrator, you can refer to Kubernetes documentation to configure it using [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
 
-### RULE \#5 - Disable inter-container communication (--icc=false)
+### RULE \#5 - Be mindful of Inter Container Connectivity
```
Comment (szh) [inline cheatsheets/Docker_Security_Cheat_Sheet.md:136]:
```suggestion
### RULE \#5 - Be mindful of Inter-Container Connectivity
```
I think a hyphen is called for here as well. It's hyphenated in the [docs](https://docs.docker.com/network/drivers/bridge/#connection-limit-for-bridge-networks)

File: cheatsheets/Docker_Security_Cheat_Sheet.md [hunk 4]
```diff
@@ -250,10 +248,10 @@ To detect misconfigurations in Docker:
 
 By default, the Docker daemon is configured to have a base logging level of 'info', and if this is not the case: set the Docker daemon log level to 'info'. Rationale: Setting up an appropriate log level, configures the Docker daemon to log events that you would want to review later. A base log level of 'info' and above would capture all logs except the debug logs. Until and unless required, you should not run docker daemon at the 'debug' log level.
 
-To configure the log level in docker-compose:
+To configure the log level in Docker Compose:
 
 ```bash
-docker-compose --log-level info up
+docker compose --log-level info up -d
```
Comment (szh) [inline cheatsheets/Docker_Security_Cheat_Sheet.md:254]: Why add `-d`?

File: cheatsheets/Docker_Security_Cheat_Sheet.md [hunk 5]
```diff
@@ -287,6 +285,33 @@ Rootless mode graduated from experimental in Docker Engine v20.10 and should be
...
+    my_secret:
+      file: ./super-secret-data.txt
+  services:
+    web:
+      image: nginx:latest
+      secrets:
+        - my_secret
+```
+
+While Docker Secrets are generally, this approach is not recommended for Kubernetes, where secrets are stored in plaintext by default. In Kubernetes, consider using additional security measures such as etcd encryption, or third-party tools. Refer to the [Secrets Management Cheat Sheet](Secrets_Management_Cheat_Sheet.md) for more information.
```
Comment (szh) [inline cheatsheets/Docker_Security_Cheat_Sheet.md:311]: generally what?

**Review by szh — CHANGES_REQUESTED** [review]
Overall I really like this. This is a much needed update and also improves the grammar of the cheat sheet. I left a few comments and suggestions, feel free to let me know if you disagree with any of them.

File: cheatsheets/Docker_Security_Cheat_Sheet.md (hunk 4 above)
Comment (otkd) [reply cheatsheets/Docker_Security_Cheat_Sheet.md]: Added detached flag for a slightly more typical example, but taking a second look at that section the whole command isn't actually relevant.

**Comment by otkd** [comment]
> Overall I really like this. This is a much needed update and also improves the grammar of the cheat sheet. I left a few comments and suggestions, feel free to let me know if you disagree with any of them.

@szh agree with all of them, also prompted me to take a look at some of the other sections which I've updated as well

**Review by jmanico — APPROVED** [review]

**Pull request merged**